modules/dhcp_common/server.rb

```diff
@@ -166,21 +167,24 @@ def find_similar_records(subnet_address, ip_address, mac_address)
     end
 
     def clean_up_add_record_parameters(in_options)
-      to_return = in_options.dup
+      options = in_options.dup
+
+      options.delete("captures")
+      options.delete("splat")
 
-      to_return.delete("captures")
-      to_return.delete("splat")
+      ip = options.delete("ip")
+      mac = options.delete("mac")
 
-      ip = to_return.delete("ip")
-      mac = to_return.delete("mac")
+      name = options.delete("name")
+      hostname = options.delete("hostname")
 
-      name = to_return.delete("name")
-      hostname = to_return.delete("hostname")
+      options.delete("subnet") # Not a valid key; remove it to prevent conflict with :subnet
+      subnet = options.delete("network")
 
-      to_return.delete("subnet") # Not a valid key; remove it to prevent conflict with :subnet
-      subnet = to_return.delete("network")
+      # convert string keys to symbols
+      options = options.inject({}){|all, current| all[current[0].to_sym] = current[1]; all}
```

Comment:
If I don't misread the diff, this line is the only relevant change.
Wouldn't it be cleaner to first convert all the keys to symbols and then extract the relevant data (ip, mac, name, hostname, ...)?

The ticket states, this was related to #548 - is it?

Reply:
> Wouldn't it be cleaner to first convert all the keys to symbols and then extract the relevant data (ip, mac, name, hostname, ...)?

I don't think it makes any difference, tbh.

> The ticket states, this was related to #548 - is it?

I don't think so.

Reply:
Well, I was doing git bisect and it pointed me on that commit. I *think* the issue was that it made additional copy (merge call) instead passing in as reference.

Anyway, I think we should be defensive in this method and accept keys both as strings and symbols - I'd also prefer "convert first, use symbols everywhere" approach to be honest, possibly can be slower during DoS but that should not be huge problem.